semver.go

```diff
@@ -243,7 +243,11 @@ func ParseTolerant(s string) (Version, error) {
 	// Remove leading zeros.
 	for i, p := range parts {
 		if len(p) > 1 {
-			parts[i] = strings.TrimPrefix(p, "0")
+			p = strings.TrimLeft(p, "0")
+			if len(p) == 0 || !strings.ContainsAny(p[0:1], "0123456789") {
```

Comment:
might compare to empty string and drop `0` such that

`if p == "" || !strings.ContainsAny(p[:1], "123456789) {`